src/main/java/edu/ucsb/mapache/controllers/MessagesController.java

```diff
@@ -0,0 +1,68 @@
+package edu.ucsb.mapache.controllers;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.validation.Valid;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import edu.ucsb.mapache.advice.AuthControllerAdvice;
+import edu.ucsb.mapache.documents.Channel;
+import edu.ucsb.mapache.documents.Message;
+import edu.ucsb.mapache.documents.SlackUser;
+import edu.ucsb.mapache.entities.Admin;
+import edu.ucsb.mapache.entities.AppUser;
+import edu.ucsb.mapache.repositories.AppUserRepository;
+import edu.ucsb.mapache.repositories.ChannelRepository;
+import edu.ucsb.mapache.repositories.MessageRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@RestController
+@RequestMapping("/api/members/messages")
+public class MessagesController {
+    private final Logger logger = LoggerFactory.getLogger(MessagesController.class);
+
+    @Autowired
+    ChannelRepository channelRepository;
+    @Autowired
+    MessageRepository messageRepository;
+
+    @Autowired
+    private AuthControllerAdvice authControllerAdvice;
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    private ResponseEntity<String> getUnauthorizedResponse(String roleRequired) throws JsonProcessingException {
+        Map<String, String> response = new HashMap<String, String>();
+        response.put("error", String.format("Unauthorized; only %s may access this resource.", roleRequired));
+        String body = mapper.writeValueAsString(response);
+        return new ResponseEntity<String>(body, HttpStatus.UNAUTHORIZED);
+    }
+
+    @GetMapping("/contentsearch")

```

Comment:
It seems that the search is not of a specific channel. The name of this method could be misleading.